app/assets/javascripts/initializers/initializeDiscussionLockButtons.js

```diff
@@ -0,0 +1,309 @@
+function initializeDiscussionLockButtons() {
+  const articleContainer = document.getElementById('article-show-container');
+  if (articleContainer) {
+    const isDiscussionLocked = articleContainer.dataset.discussionLocked;
+
+    if (isDiscussionLocked == 'true') {
+      displayDiscussionLockReason();
+      hideElementById('new_comment');
+    } else {
+      showElementById('discussion-lock-action');
+    }
+
+    displayButtons(articleContainer);
+  }
+  addConfirmationModalClickHandlers();
+}
+
+function displayDiscussionLockReason(reason) {
+  const discussionLockReason = document.getElementById(
+    'discussion-lock-reason',
+  );
+  if (discussionLockReason) {
+    discussionLockReason.classList.remove('hidden');
+
+    const discussionLockReasonText = document.getElementById(
+      'discussion-lock-reason-text',
+    );
+    if (reason && discussionLockReasonText) {
+      discussionLockReasonText.textContent = reason;
+    }
+  }
+}
+
+function hideDiscussionLockReason() {
+  const discussionLockReason = document.getElementById(
+    'discussion-lock-reason',
+  );
+  if (discussionLockReason) {
+    discussionLockReason.classList.add('hidden');
+  }
+
+  const discussionLockReasonText = document.getElementById(
+    'discussion-lock-reason-text',
+  );
+  if (discussionLockReasonText) {
+    discussionLockReasonText.textContent = '';
+  }
+}
+
+function displayButtons() {
+  var user = userData();
+  var articleContainer = document.getElementById('article-show-container');
+  if (user && articleContainer) {
+    var authorId = parseInt(articleContainer.dataset.authorId, 10);
+
+    if (user.admin || authorId === user.id) {
+      var discussionLockButts = document.getElementsByClassName(
+        'discussion-lock-actions',
+      );
+
+      for (let i = 0; i < discussionLockButts.length; i += 1) {
+        let butt = discussionLockButts[i];
+        butt.classList.remove('hidden');
+      }
+    }
+  }
+}
+
+function addConfirmationModalClickHandlers() {
+  const discussionLockButton = document.getElementById('discussion-lock-btn');
+  if (discussionLockButton) {
+    discussionLockButton.addEventListener('click', function (e) {
+      showElementById('discussion-lock-confirmation-modal');
+    });
+  }
```

Comment:
For modals to be accessible, we need to make sure focus is transferred and trapped in the modal content / transferred back when it closes. We have a helper that's globally accessible to this: `Forem.showModal()`

You can find some examples of this in use in `app/assets/javascripts/utilities/showLoginModal.js` and `app/assets/javascripts/utilities/showUserAlertModal.js`. 

Give me a shout if you want to discuss how it works or anything - it will mean removing some of the HTML you've added in `_discussion_lock.html.erb `, just because the `showModal` function will automatically create the modal container and header with title/close icon.